micrometer-api/src/main/java/io/micrometer/api/instrument/observation/SimpleObservation.java

```diff
@@ -0,0 +1,130 @@
+package io.micrometer.api.instrument.observation;
+
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.stream.Collectors;
+
+import io.micrometer.api.instrument.Tag;
+import io.micrometer.api.lang.Nullable;
+
+public class SimpleObservation implements Observation {
+    private final ObservationRegistry registry;
+    @Nullable private Throwable error;
+    private final Context context;
+    @SuppressWarnings("rawtypes")
+    private final Deque<ObservationHandler> handlers;
+
+    public SimpleObservation(String name, ObservationRegistry registry) {
+        this(name, registry, new Context());
+    }
+
+    public SimpleObservation(String name, ObservationRegistry registry, Context context) {
+        this.registry = registry;
+        this.context = context.setName(name);
+        this.handlers = registry.config().getObservationHandlers().stream()
+                .filter(handler -> handler.supportsContext(this.context))
+                .collect(Collectors.toCollection(ArrayDeque::new));
+    }
+
+    @Override
+    public Observation displayName(String displayName) {
+        this.context.setDisplayName(displayName);
+        return this;
+    }
+
+    @Override
+    public Observation lowCardinalityTag(Tag tag) {
+        this.context.addLowCardinalityTag(tag);
+        return this;
+    }
+
+    @Override
+    public Observation highCardinalityTag(Tag tag) {
+        this.context.addHighCardinalityTag(tag);
+        return this;
+    }
+
+    @Override
+    public Observation error(Throwable error) {
+        this.error = error;
+        this.notifyOnError();
+        return this;
+    }
+    @Override
+    public Observation start() {
+        this.notifyOnObservationStarted();
+        return this;
+    }
+
+    @Override
+    public void stop() {
+        this.notifyOnObservationStopped();
+    }
+
+    @Override
+    public Scope openScope() {
+        Scope scope = new SimpleScope(this.registry, this);
+        this.notifyOnScopeOpened();
+        return scope;
+    }
+
+    @Override
+    public String toString() {
+        return "{"
+                + "name=" + this.context.getName() + "(" + this.context.getDisplayName() + ")"
+                + ", error=" + this.error
+                + ", context=" + this.context
+                + '}';
+    }
+
+    @SuppressWarnings("unchecked")
+    private void notifyOnObservationStarted() {
+        this.handlers.forEach(handler -> handler.onStart(this, this.context));
```

Comment:
Why are we passing the observation? Now it's kind of empty, what would the handler use it for?